app/routes.py

```diff
@@ -1,2 +1,297 @@
-from flask import Blueprint
+
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify
+import datetime
+import os 
+import requests
+
+goal_bp = task_bp = Blueprint("goal", __name__, url_prefix="/goals")
+
+@goal_bp.route("", methods=["GET", "POST"], strict_slashes=False)
+
+def handle_goal():
+
+    if request.method == "GET":
+
+        goals = Goal.query.all()
+        response = []
+
+        if not goals:
+            return jsonify(response), 200
+
+
+        for goal in goals:
+            response.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+        
+        return jsonify(response), 200
+    
+    elif request.method == "POST":
+
+        request_body = request.get_json()
+
+        if request_body:
+            new_goal = Goal(title=request_body["title"])
+
+            db.session.add(new_goal)
+            db.session.commit()
+
+            response = {"goal": new_goal.json_response()}
+
+            return jsonify(response), 201
+        
+        else: 
+            response = {"details": "Invalid data"}
+            return jsonify(response), 400
+
+
+@goal_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+
+def get_one_goal(goal_id):
```

Comment:
Generally with decorators we will not put a blank line between the them and function being decorated. 

```suggestion
@goal_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
def get_one_goal(goal_id):
```